internal/flavors/publisher.go

```diff
@@ -41,6 +41,8 @@ type Publisher struct {
 	interval  time.Duration
 	threshold int
 	client    client
+
+	eventsBuffer []beat.Event
```

Comment:
As a note, this changes `Publisher` from being thread-safe to being thread-unsafe

Reply:
I think we use the `HandleEvents` from a single go routine both on posture and vuln, I did n't find any case where we use it as a worker and spawn multiple go routines using the same struct instance. 

Also there is no other way to test the events buffer capacity growth and reset, unless you either provide the buffer externally (which might be a bit ugly API wise) or put the buffer in the struct's state.  


So to me that's fine since there is no need to do otherwise. But we can explicitly note it with a comment perhaps.

Reply:
Yes, sorry should have been clearer: I don't see any issue, just wanted to explicitly state in my review.